infrastructure/src/main/java/org/corfudb/infrastructure/log/FileSystemAgent.java

```diff
@@ -71,7 +71,7 @@ private long getLogSizeLimit() {
 
         // Derived size in bytes that normal writes to the log unit are capped at.
         // This is derived as a percentage of the log's filesystem capacity.
-        return (long) (fileSystemCapacity * config.limitPercentage / 100);
+        return (long) ((fileSystemCapacity - config.partitionOffset * 1000) * config.limitPercentage / 100);
```

Comment:
Does this assume that Corfu always has the last partition?

Also, it seems like `fileSystemCapacity` is the actual partition size, but `config.partitionOffset` assumes that `fileSystemCapacity` is the total drive/volume size. Is that correct?

Reply:
No, partitionOffset assumes fileSystemCapacity as the actual partition size (size of the lodir's parent).
And I am not sure if Corfu always has the last partition.

Reply:
I might be missing something, then the `fileSystemCapacity - config.partitionOffset * 1000` calculation doesn't make sense. 

If the capacity is 100 and the partition offset is 30, then 70 is what is available only if partitionOffset is the last partition. For the following case the calculation would be wrong:

0 | partition offset1 | partition offset2(corfu) | partition offset3 | 100

The calculation would subsume partition offset3 which doesn't seem like the desired outcome. Also, the partition offset calculation doesn't seem to make sense unless you are subtracting it from the volume/device total size and not the partition size.

Reply:
Yes, you are right. 
My intention was to actually do the calculation based on the entire volume size but misunderstood the terms here. 
Looks like this change may not be required even if the volume is used by another process, since we anyway calculate `limit` based on `fileSystemCapacity` (which is the actual partition size)

Reply:
Does `fileSystemCapacity` actually capture the partition size, or is it the volume size? The variable name might be a little misleading.

Reply:
It captures the parent dir of the logdir param passed. So this PR is not required for our usecase.

Reply:
Reopening this PR - 
I overlooked the method `getFileSystemCapacity()` previously - `fileSystemCapacity` is the total space of the entire filesystem as per the line -`Files.getFileStore(corfuDir).getTotalSpace();`

Usecase: 
We want to store other data irrelevant to corfu in the same volume. 
For example, we want to have `/root/corfu-data` and `/root/not-corfu-data`, where `not-corfu-data` will be using x amount of data as passed through the param `orthogonalData`.  So we want to calculate read_only threshold for corfu-data subtracting the allocated data in not-corfu-data.